File: 2019-01-16/W2D3-PassingData/W2D3-PassingData/ViewController.m
//
//  ViewController.m
//  W2D3-PassingData
//
//  Created by Jason Liang on 1/16/19.
//  Copyright © 2019 jasonliang. All rights reserved.
//

#import "ViewController.h"
#import "DetailViewController.h"

@interface ViewController ()


@end

@implementation ViewController

- (void)viewDidLoad {
  [super viewDidLoad];
}

- (IBAction)submit {
  [self performSegueWithIdentifier:@"submit" sender:nil];
}

- (void)prepareForSegue:(UIStoryboardSegue *)segue sender:(id)sender {
  
  if ([segue.identifier isEqualToString:@"submit"]) {
    DetailViewController *dvc = (DetailViewController *)segue.destinationViewController;  
    dvc.textInfo = self.textField.text;
    
  }
}


@end

File: 2017-03-01/W8D3DemoWidget/W8D3Demo/ViewController.swift
//
//  ViewController.swift
//  W8D3Demo
//
//  Created by Jason Liang on 3/1/17.
//  Copyright © 2017 Jason Liang. All rights reserved.
//

import UIKit
import W8D3DemoKit

class ViewController: UIViewController {

  @IBOutlet weak var resultLabel: UILabel!
  @IBOutlet weak var addButton: UIButton!
  @IBOutlet weak var textField: UITextField!
  
  let manager = DemoKitManager()
  
  override func viewDidLoad() {
    super.viewDidLoad()
    let container = DemoHelper.sharedInstance.persistentContainer
    print("container \(container)")
  }
  
  func refresh() {
    resultLabel.text = "\(manager.getList().count)"
    let ud = UserDefaults(suiteName: "group.com.jasonliang.demo")
    ud?.set(true, forKey: "key")
    ud?.synchronize()
  }

  @IBAction func addButtonTapped(_ sender: Any) {
    manager.addEntry(name: "Jason")
    refresh()
  }

}

